crates/ruff/src/rules/ruff/rules/collection_literal_concatenation.rs

```diff
@@ -54,80 +54,135 @@ enum Kind {
-                    elts: l_elts,
-                    ctx,
-                    range: _,
-                }),
-                _,
-            ) => (Kind::List, right, l_elts, false, ctx),
-            (
-                Expr::Tuple(ast::ExprTuple {
-                    elts: l_elts,
-                    ctx,
-                    range: _,
-                }),
-                _,
-            ) => (Kind::Tuple, right, l_elts, false, ctx),
-            (
-                _,
-                Expr::List(ast::ExprList {
-                    elts: r_elts,
-                    ctx,
-                    range: _,
-                }),
-            ) => (Kind::List, left, r_elts, true, ctx),
-            (
-                _,
-                Expr::Tuple(ast::ExprTuple {
-                    elts: r_elts,
-                    ctx,
-                    range: _,
-                }),
-            ) => (Kind::Tuple, left, r_elts, true, ctx),
-            _ => return,
-        };
+    let (kind, splat_element, other_elements, splat_at_left, ctx) = match (&new_left, &new_right) {
+        (
+            Expr::List(ast::ExprList {
+                elts: l_elts,
+                ctx,
+                range: _,
+            }),
+            _,
+        ) => (Kind::List, &new_right, l_elts, false, ctx),
+        (
+            Expr::Tuple(ast::ExprTuple {
+                elts: l_elts,
+                ctx,
+                range: _,
+            }),
+            _,
+        ) => (Kind::Tuple, &new_right, l_elts, false, ctx),
+        (
+            _,
+            Expr::List(ast::ExprList {
+                elts: r_elts,
+                ctx,
+                range: _,
+            }),
+        ) => (Kind::List, &new_left, r_elts, true, ctx),
+        (
+            _,
+            Expr::Tuple(ast::ExprTuple {
+                elts: r_elts,
+                ctx,
+                range: _,
+            }),
+        ) => (Kind::Tuple, &new_left, r_elts, true, ctx),
+        _ => return None,
+    };
 
     // We'll be a bit conservative here; only calls, names and attribute accesses
     // will be considered as splat elements.
-    if !(splat_element.is_call_expr()
+    let mut new_elts;
+    if splat_element.is_call_expr()
         || splat_element.is_name_expr()
-        || splat_element.is_attribute_expr())
+        || splat_element.is_attribute_expr()
     {
-        return;
+        new_elts = make_splat_elts(splat_element, other_elements, splat_at_left);
+    }
+    // If the splat element is itself a list/tuple, insert them in the other list/tuple.
+    else if splat_element.is_list_expr() || splat_element.is_tuple_expr() {
+        new_elts = other_elements.clone();
+
+        let mut elts = match splat_element {
+            Expr::List(ast::ExprList { elts, .. }) => elts.clone(),
+            Expr::Tuple(ast::ExprTuple { elts, .. }) => elts.clone(),
+            _ => return None,
+        };
+
+        if splat_at_left {
```

Comment:
In what case would `splat_at_left` be true here?

Reply:
It seems like it's always true indeed, I've removed the `if`, thank you!